src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/RaiderWalkAI.java

```diff
@@ -74,6 +91,32 @@ private boolean walk()
                 final BlockPos moveToPos = ShipBasedRaiderUtils.chooseWaypointFor(wayPoints, raider.blockPosition(), targetBlock);
                 raider.getNavigation().moveToXYZ(moveToPos.getX(), moveToPos.getY(), moveToPos.getZ(), !moveToPos.equals(targetBlock) && moveToPos.distManhattan(wayPoints.get(0)) > 50 ? 1.8 : 1.1);
             }
+            else if (walkInBuildingState)
+            {
+                final BlockPos moveToPos = findRandomPositionToWalkTo();
+                if (moveToPos != null)
+                {
+                    raider.getNavigation().moveToXYZ(moveToPos.getX(), moveToPos.getY(), moveToPos.getZ(), 0.9);
+                    if (raider.blockPosition().distSqr(moveToPos) < 10)
+                    {
+                        if (raider.getRandom().nextDouble() < 0.25)
+                        {
+                            walkInBuildingState = false;
+                            targetBlock = null;
+                        }
+                        else
+                        {
+                            randomPathResult = null;
+                            findRandomPositionToWalkTo();
+                        }
+                    }
+                }
+            }
+            else if (raider.blockPosition().distSqr(targetBlock) < 25)
+            {
+                findRandomPositionToWalkTo();
```

Comment:
should probably set a short walk timeout here aswell,  walkTimer = raider.level.getGameTime() + TICKS_SECOND * 30;
or so

Reply:
fair, but probably only if walkTimer is > than that

Reply:
think you can use it anyways, walk timer is intended as timeout for the current walking action, so you set the one you want as timeout for in building walk